Generate serialization for some PassKit related classes

beidson opened the pull request:
#### 61c913ecf3c525a7de3662d08da273cbca20b362
<pre>
Generate serialization for some PassKit related classes
<a href="https://bugs.webkit.org/show_bug.cgi?id=268960">https://bugs.webkit.org/show_bug.cgi?id=268960</a>
<a href="https://rdar.apple.com/122515353">rdar://122515353</a>

Reviewed by Alex Christensen.

This started as an exercise to generate `PKPayment` and `PKPaymentMethod`
It turns out doing those two brought in even more PK types and some non-PK types.

I generate many of them, but had to call it quits at some point, adding some special
cases where we have a known type that we put through the generic NSSecureCoding path.

Generating for those types will come soon after.

* Source/WebCore/PAL/pal/cocoa/ContactsSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/ContactsSoftLink.mm:
* Source/WebCore/PAL/pal/cocoa/PassKitSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/PassKitSoftLink.mm:
* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::getClass&lt;CNContact&gt;):
(IPC::getClass&lt;PKPayment&gt;):
(IPC::getClass&lt;PKPaymentToken&gt;):
(IPC::getClass&lt;PKShippingMethod&gt;):
(IPC::getClass&lt;PKDateComponentsRange&gt;):
(IPC::getClass&lt;PKPaymentMethod&gt;):
(IPC::typeFromObject):
(IPC::shouldEnableStrictMode):
(IPC::decodeObjectDirectlyRequiringAllowedClasses&lt;NSObject&lt;NSSecureCoding&gt;&gt;):
* Source/WebKit/Shared/Cocoa/CoreIPCContacts.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.mm:
(WebKit::valueFromID):
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:
* Source/WebKit/Shared/Cocoa/WKKeyedCoder.mm:
(-[WKKeyedCoder encodeBool:forKey:]):
(-[WKKeyedCoder decodeBoolForKey:]):
(-[WKKeyedCoder encodeInt64:forKey:]):
(-[WKKeyedCoder decodeInt64ForKey:]):
(-[WKKeyedCoder encodeInteger:forKey:]):
(-[WKKeyedCoder decodeIntegerForKey:]):
* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(wkDDActionContext_isEqual):
(wkSecureCoding_isEqual):
(operator==):
(postalAddressForTesting):
(pkContactForTesting):
(TEST):
(isEqual): Deleted.

Canonical link: <a href="https://commits.webkit.org/274384@main">https://commits.webkit.org/274384@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/b31ae14888d8552aa9d32ba626f9361671138ae8

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/38928 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/48/builds/17859 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/41276 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/41460 "Built successfully") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/32/builds/34645 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/41234 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/49/builds/20725 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/51/builds/15208 "Built successfully") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/34/builds/32612 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/39501 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/47/builds/15032 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/33756 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/13059 "Passed tests") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/42/builds/13023 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/34677 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/42737 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/35331 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/35019 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/38850 "Passed tests") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/44/builds/13739 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/50/builds/11335 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/37084 "Passed tests") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/15346 "Built successfully") | | | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/8718 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/15007 "Built successfully") | | | 
| | [  ~~🛠 watch-sim~~](https://ews-build.webkit.org/#/builders/45/builds/14832 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | | | 
<!--EWS-Status-Bubble-End-->

achristensen07 commented on Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
```diff
@@ -417,20 +424,43 @@ inline bool isEqual(WKDDActionContext *a, WKDDActionContext* b)
+        // These classes do not have isEqual: methods useful for our unit testing, so we'll swap it in ourselves.
+
+#if USE(PASSKIT) && !PLATFORM(WATCHOS) && !PLATFORM(APPLETV)
+        class_addMethod(PAL::getPKPaymentMethodClass(), @selector(isEqual:), (IMP)wkSecureCoding_isEqual, "v@:@");
```
This looks awful.  Can't we just check the class and call a function like normal people?

beidson replied:
I would not propose all this for production code.
I'm very happy doing it in testing code. =D

beidson committed:
Generate serialization for some PassKit related classes
https://bugs.webkit.org/show_bug.cgi?id=268960
rdar://122515353

Reviewed by Alex Christensen.

This started as an exercise to generate `PKPayment` and `PKPaymentMethod`
It turns out doing those two brought in even more PK types and some non-PK types.

I generate many of them, but had to call it quits at some point, adding some special
cases where we have a known type that we put through the generic NSSecureCoding path.

Generating for those types will come soon after.

* Source/WebCore/PAL/pal/cocoa/ContactsSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/ContactsSoftLink.mm:
* Source/WebCore/PAL/pal/cocoa/PassKitSoftLink.h:
* Source/WebCore/PAL/pal/cocoa/PassKitSoftLink.mm:
* Source/WebCore/PAL/pal/spi/cocoa/PassKitSPI.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.h:
* Source/WebKit/Shared/Cocoa/ArgumentCodersCocoa.mm:
(IPC::getClass<CNContact>):
(IPC::getClass<PKPayment>):
(IPC::getClass<PKPaymentToken>):
(IPC::getClass<PKShippingMethod>):
(IPC::getClass<PKDateComponentsRange>):
(IPC::getClass<PKPaymentMethod>):
(IPC::typeFromObject):
(IPC::shouldEnableStrictMode):
(IPC::decodeObjectDirectlyRequiringAllowedClasses<NSObject<NSSecureCoding>>):
* Source/WebKit/Shared/Cocoa/CoreIPCContacts.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.h:
* Source/WebKit/Shared/Cocoa/CoreIPCNSCFObject.mm:
(WebKit::valueFromID):
* Source/WebKit/Shared/Cocoa/CoreIPCPassKit.serialization.in:
* Source/WebKit/Shared/Cocoa/WKKeyedCoder.mm:
(-[WKKeyedCoder encodeBool:forKey:]):
(-[WKKeyedCoder decodeBoolForKey:]):
(-[WKKeyedCoder encodeInt64:forKey:]):
(-[WKKeyedCoder decodeInt64ForKey:]):
(-[WKKeyedCoder encodeInteger:forKey:]):
(-[WKKeyedCoder decodeIntegerForKey:]):
* Tools/TestWebKitAPI/Tests/IPC/IPCSerialization.mm:
(wkDDActionContext_isEqual):
(wkSecureCoding_isEqual):
(operator==):
(postalAddressForTesting):
(pkContactForTesting):
(TEST):
(isEqual): Deleted.

Canonical link: https://commits.webkit.org/274384@main
